### DISCUSSION TITLE
[v5] Server hooks

### DISCUSSION THREAD

**Discussion by artem-a** [adonisjs/core#2519] [Help]
Hi there. Where can I read about server hooks? Does v5 have the same flow of V4 for adding and using the Ignitor#hooks?

Thanks for help.

**Comment by fareeda0** [comment]
What exactly are you trying to accomplish?

You should be able to achieve the same in v5 using a provider.

**Reply by artem-a** [reply]
No, I need to know when the server will start or has been started.

In V4 I can do this

```javascript
const { hooks } = require('@adonisjs/ignitor')

hooks.before.httpServer(async () => {
  .....
})
```

**Reply by thetutlage** [reply]
You can use the provider as @fareeda0 said, and write your code inside the `ready` method
